wcfsetup/install/files/lib/system/session/SessionHandler.class.php

```diff
@@ -991,4 +984,148 @@ protected function getSpiderID($userAgent) {
 	public function isFirstVisit() {
 		return $this->firstVisit;
 	}
+	
+	/**
+	 * Returns all user sessions for a specific user.
+	 *
+	 * @return      Session[]
+	 * @throws      \InvalidArgumentException if the given user is a guest.
+	 * @since       5.4
+	 */
+	public function getUserSessions(User $user): array {
+		return $this->getSessions($user);
+	}
+	
+	/**
+	 * Returns all acp sessions for a specific user.
+	 * 
+	 * @return      Session[]
+	 * @throws      \InvalidArgumentException if the given user is a guest.
+	 * @since       5.4
+	 */
+	public function getAcpSessions(User $user): array {
+		return $this->getSessions($user, true);
+	}
+	
+	/**
+	 * Returns all sessions for a specific user.
+	 * 
+	 * @return      Session[]
+	 * @throws      \InvalidArgumentException if the given user is a guest.
+	 * @since       5.4
+	 */
+	private function getSessions(User $user, bool $isAcp = false): array {
+		if (!$user->userID) {
+			throw new \InvalidArgumentException("The given user is a guest.");
+		}
+		
+		$sql = "SELECT          *
+			FROM            wcf".WCF_N."_". ($isAcp ? 'acp' : 'user') ."_session
+			WHERE           userID = ?";
+		$statement = WCF::getDB()->prepareStatement($sql);
+		$statement->execute([$user->userID]);
+		
+		$sessions = [];
+		while ($row = $statement->fetchArray()) {
+			$sessions[] = new Session($row, $isAcp);
+		}
+		
+		return $sessions;
+	}
+	
+	/**
+	 * Deletes the user sessions for a specific user, except the session with the given session id.
+	 * 
+	 * If the given session id is `null` or unknown, all sessions of the user will be deleted.
+	 * 
+	 * @throws      \InvalidArgumentException if the given user is a guest.
+	 * @since       5.4
+	 */
+	public function deleteUserSessionsExcept(User $user, ?string $sessionID = null): void {
+		$this->deleteSessionsExcept($user, $sessionID);
+	}
+	
+	/**
+	 * Deletes the acp sessions for a specific user, except the session with the given session id.
+	 * 
+	 * If the given session id is `null` or unknown, all acp sessions of the user will be deleted.
+	 * 
+	 * @throws      \InvalidArgumentException if the given user is a guest.
+	 * @since       5.4
+	 */
+	public function deleteAcpSessionsExcept(User $user, ?string $sessionID = null): void {
+		$this->deleteSessionsExcept($user, $sessionID, true);
+	}
+	
+	/**
+	 * Deletes the sessions for a specific user, except the session with the given session id.
+	 *
+	 * If the given session id is `null` or unknown, all acp sessions of the user will be deleted.
+	 *
+	 * @throws      \InvalidArgumentException if the given user is a guest.
+	 * @since       5.4
+	 */
+	private function deleteSessionsExcept(User $user, ?string $sessionID = null, bool $isAcp = false): void {
```

Comment:
```suggestion
	private function deleteSessionsExcept(User $user, ?string $sessionID, bool $isAcp): void {
```

Not putting default values here makes it harder to mis-use this method.